website/addons/dataverse/templates/dataverse_credentials_modal.mako

```diff
@@ -0,0 +1,81 @@
+<div id="dataverseInputCredentials" class="modal fade">
+    <div class="modal-dialog modal-lg">
+        <div class="modal-content">
+
+            <div class="modal-header">
+                <h3>Connect a Dataverse Account</h3>
+            </div>
+
+            <form>
+                <div class="modal-body">
+
+                    <div class="row">
+
+                        <div class="col-sm-6">
+
+                            <!-- Select Dataverse installation -->
+                            <div class="form-group">
+                                <label for="hostSelect">Dataverse Repository</label>
+                                <select class="form-control"
+                                        id="hostSelect"
+                                        data-bind="options: visibleHosts,
+                                                   optionsCaption: 'Select an item...',
+                                                   value: selectedHost">
+                                </select>
+                            </div>
+
+                            <!-- Custom input -->
+                            <div data-bind="if: useCustomHost">
+                                <div class="input-group">
+                                    <div class="input-group-addon">https://</div>
+                                    <input type="text" class="form-control" name="customHost" data-bind="value: customHost">
+                                </div>
+                                <div class="text-info" style="text-align: center">
+                                    <em>Only Dataverse repositories v4.0 or higher are supported.</em>
+                                </div>
+                            </div>
+
+                        </div>
+
+                        <div class="col-sm-6">
+
+                            <!-- API Token Input-->
+                            <div class="form-group" data-bind="if: showApiTokenInput">
+                                <label for="apiToken">
+                                    API Token
+                                    <!-- Link to API token generation page -->
+                                    <a href="{{tokenUrl}}"
```

Comment:
To prevent spiders from crawling invalid URLs (like '{{tokenUrl}}'), use `data-bind="attr: {href: tokenURL}"`. 

Please change this in other places where applicable.